### PR TITLE
rbac: add unit tests for matchers to increase coverage

### PR DESCRIPTION
**Pull request by agrawroh**
## Description

This PR add some more unit tests to improve the code coverage of RBAC matchers class.

---

**Commit Message:** rbac: add unit tests for matchers to increase coverage
**Additional Description:** Added some more unit tests to improve the code coverage of RBAC matchers class.
**Risk Level:** Very Low
**Testing:** Added Unit Tests
**Docs Changes:** N/A
**Release Notes:** N/A

### CODE REVIEW

File: test/extensions/filters/common/rbac/matchers_test.cc [hunk 1]
```diff
@@ -699,6 +699,263 @@ TEST(MetadataMatcher, SourcedMetadataMatcher) {
...
+  headers.setReference(header_name, "value1");
+
+  RBAC::HeaderMatcher matcher(config, factory_context);
+  checkMatcher(matcher, true, Envoy::Network::MockConnection(), headers);
+
+  // Test with non-matching value
+  headers.remove(header_name);
+  headers.setReference(header_name, "value2");
+  checkMatcher(matcher, false, Envoy::Network::MockConnection(), headers);
+}
```
Comment (wbpcode) [inline test/extensions/filters/common/rbac/matchers_test.cc:774]: Where is the multiple header values matching?

**Review by wbpcode — COMMENTED** [review]
Thanks for this improvement. Only one comment is added.

File: test/extensions/filters/common/rbac/matchers_test.cc (hunk 1 above)
Comment (agrawroh) [reply test/extensions/filters/common/rbac/matchers_test.cc]: @wbpcode Thanks for the feedback! I tried adding some cases.

**Review by wbpcode — APPROVED** [review]

**Pull request merged**